src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -55,27 +58,69 @@ const useFetchChannelHook = () => {
     newChannel.channel = {...channel};
     const channelType = channel?.type;
 
-    try {
-      queue.addJob({
-        label: `saveChannelData-${channel?.id}`,
-        task: () => {
-          return new Promise((resolve) => {
-            const channelList = ChannelList.fromChannelAPI(
-              newChannel,
-              type[channelType],
-              undefined,
-              anonUserInfo
-            );
-
-            channelList.saveIfLatest(localDb).then(() => {
-              refresh('channelList');
-              resolve(true);
+    if (channelType === 'topics') {
+      try {
+        const cleanTopicName = newChannel?.id?.replace('topic_', '');
+        const response = await getLatestTopicPost(cleanTopicName);
+
+        if (!response || response?.status !== 'success') return null;
+
+        newChannel.firstMessage = {
+          text: response?.message,
+          message: response?.message,
+          textOwnMessage: response?.message
+        };
+        newChannel.topicPostExpiredAt = response?.expired_at;
+        newChannel.updated_at = response?.time;
+        newChannel.created_at = response?.time;
+        newChannel.unreadCount = response?.unread_count;
+
+        queue.addPriorityJob({
+          id: `saveChannelData-${channel?.id}`,
+          priority: QueueJobPriority.MEDIUM,
+          operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveTopicChannel,
+          label: `saveChannelData-${channel?.id}`,
+          task: () => {
+            return new Promise((resolve) => {
+              const channelList = ChannelList.fromChannelAPI(
+                newChannel,
+                type[channelType],
+                undefined,
+                anonUserInfo
+              );
+
+              channelList.saveIfLatest(localDb).then(() => {
+                refresh('channelList');
+                resolve(true);
+              });
             });
-          });
-        }
-      });
-    } catch (e) {
-      console.log('error on helperChannelPromiseBuilder', e);
+          }
+        });
+      } catch (e) {
+        console.log('error on helperChannelPromiseBuilder', e);
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This new block of code handles the case when the channel type is 'topics'. It fetches the latest topic post and updates the channel data accordingly. The error handling seems appropriate, but it would be better to provide more context in the error message for easier debugging.
```diff
        console.log('error on helperChannelPromiseBuilder', e);
+100:         console.log(`Error on helperChannelPromiseBuilder for channel ${channel?.id}`, e);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->